File: spring-framework-samples/src/main/java/org/springframework/samples/BeanInitLifeCycle.java
package org.springframework.samples;

import javax.annotation.PostConstruct;
import org.springframework.beans.factory.InitializingBean;
import org.springframework.stereotype.Component;

@Component
public class BeanInitLifeCycle implements InitializingBean {


	public BeanInitLifeCycle() {
		System.out.println("构造函数");
	}

	@PostConstruct
	public void postConstruct(){
		System.out.println("postConstruct");
	}


	public void initMethod(){
		System.out.println("init-method");
	}


	@Override
	public void afterPropertiesSet() throws Exception {
		System.out.println("afterPropertiesSet");
	}
}

File: spring-framework-samples/src/main/kotlin/org/springframework/samples/component/BeanInitLifeCycle.kt
package org.springframework.samples.component

import org.springframework.beans.factory.InitializingBean
import javax.annotation.PostConstruct

/**
 * bean初始化生命周期
 * @author wangyong
 */
class BeanInitLifeCycle : InitializingBean {

	init {
		println("构造函数")
	}

	/**
	 * 只有annotation扫描的方式才会生效
	 */
	@PostConstruct
	fun postConstruct() {
		println("postConstruct")
	}


	fun initMethod() {
		println("init-method")
	}


	override fun afterPropertiesSet() {
		println("afterPropertiesSet")
	}
	var className:String = ""


}